### DISCUSSION TITLE
Anschlussmatrix - Erläuterungen

### DISCUSSION THREAD

**Discussion by PianoRailways** [topological-state/stskit#45] [Ideas]
Einige Ideen, was man sich unter #27 vorstellen könnte:

Ein Tabellenraster welches alle Züge eines Bahnhofes auflistet, Bsp. Bern.
Fährt ein anderer Zug nach mehr als 3 Minuten nach Ankunft eines Zubringerzuges ab, so könnte dies ein Anschluss sein und das Fenster markiert das gemeinsame Kästchen im Koordinatensystem. Hat ein Zubringerzug nun Verspätung, da wird's interessant. Sagen wir, ein Anschluss wartet maximal 3-5 Minuten. Das STS-Kit setzt nun automatisch fix(5) für den Anschlusszug, was auf Gleisbelegung, Grafik etc. sichtbar wird. Ebenfalls in der Anschlussgrafik wird dies markiert, dass der Anschluss wartet. Überschreitet ein Zubringerzug diese maximale Verspätung von 5 Minuten, so wird der Anschluss nicht abgewartet.

Damit einen das System aber nicht wild irgendwelche fixes reindrückt, könnte man diese auch als "Vorschlag" nehmen, welche man akzeptieren oder ablehnen kann. Damit überlässt man auch dem FDL die Aufgabe, ob es sich um einen tatsächlichen wichtigen Anschluss oder einfach um einen Zug in die gleiche Richtung etc. handelt.

Wieder einmal eine Referenz zur RCS-Dispo Broschüre der SBB :)
![image](https://user-images.githubusercontent.com/32222604/180324808-0654dc55-6077-475e-87ec-0d33adcae28d.png)

Mit dieser Grafik könnte auch ein Interface geschaffen werden, um die Abfahrtsverspätung ankunft(), abfahrt(), fix() etc zu setzen.

**Comment by topological-state** [comment]
finde ich übrigens eine gute idee.

**Comment by topological-state** [comment]
habe jetzt mal eine basisversion einer anschlussmatrix implementiert in commit 9cae2be7

eine relation wird als anschluss gewertet, wenn zug 2 fahrplanmässig 2-15 minuten nach ankunft von zug 1 abfahren soll. die 2 minuten sind die minimale umsteigezeit. diese ist im moment fix und soll später konfigurierbar gemacht werden.
ein anschluss wird blau dargestellt, wenn die umsteigezeit erfüllt ist, ansonsten rot mit einer zahl, die die nötige verspätung angibt, damit die umsteigezeit erfüllt wird.
interaktion gibt es noch keine, ausser dass der bahnhof auf der einstellungsseite gewählt werden kann.

**Comment by topological-state** [comment]
die anschlussmatrix sieht im moment so aus:

![Screenshot from 2022-11-18 21-52-08](https://user-images.githubusercontent.com/51272421/202817976-8d9b1b22-b1cb-4b9e-814d-35a663d49e30.png)

blau: möglicher anschluss, pünktlich
rot: möglicher anschluss, umsteigezeit reicht nicht
orange: anschluss wartet (vom fdl gesetzte abhängigkeit)
grün: anschluss erfüllt, umsteigezeit um
violet: anschluss vom fdl aufgelöst
rosa: kupplung, warten auf zweiten zugteil
grau: gleicher zug


die zahl in den feldern gibt die verspätung an, die der zug haben wird, wenn er den anschluss abwartet.

die zugnamen werden nach zuggattung-farbschema eingefärbt.

**Comment by topological-state** [comment]
auf den toolbuttons sind folgende aktionen verfügbar:
- zug verbergen
- verborgene züge wieder anzeigen
- ankunft abwarten
- abfahrt abwarten
- anschluss auflösen
- anschlussstatus zurücksetzen
- wartezeit erhöhen/erniedrigen

folgende elemente sind durch klicken auswählbar:
- farbiges feld: wählt eine relation aus
- hintergrund: auswahl aufheben
- zug aus ankunfts- oder abfahrtsliste: wählt die ganze spalte bzw. zeile
